charts/argo/values.yaml

```diff
@@ -20,6 +20,15 @@ createAggregateRoles: true
 # and start argo-cli with the --namespaced flag. Use it in clusters with strict access policy.
 singleNamespace: false
 
+workflow:
+  namespace: ""  # Speficy namespace if workflows run in another namespace than argo. This controls where the service account and RBAC resources will be created.
+  serviceAccount:
+    create: false  # Specifies whether a service account should be created
+    annotations: {}
+    name: "default"  # Service account which is used to run workflows
```

Comment:
IMO maybe something like `argo-workflow` might be a better default name?

Reply:
I chose `default` because it is used per default for running workflows when not specifying `serviceAccountName` inside the workflow definition. All the example workflows provided by argo are using the `default` service account and I think those workflows will be the first new user will try. If they would need to edit the example workflows provided by argo to include a service account like `argo-workflow` first before trying them out it won't be that easy for them to get started. The official quickstart YAML is also creation a RoleBinding for `default`.

What do you think?

Reply:
Main issue here is if I try the following:

`helm upgrade --install argo charts/argo --set workflow.serviceAccount.create=true`

It results in an error:

`Unable to continue with install: ServiceAccount "default" in namespace "default" exists and cannot be imported into the current release`

So will break in that scenario, the name needs to be something other than default for that to work, IMO the examples using the default SA isn't giving the best advice and we should be pushing users to doing the right thing which is leaving the default SA alone and creating one specific to this purpose.

Reply:
Thanks for the feedback. I changed it.